assets/js/base/components/text-input/validated-text-input.tsx

```diff
@@ -39,8 +61,8 @@ const ValidatedTextInput = ( {
 		getValidationErrorId,
 	} = useValidationContext();
 
-	const textInputId = id || 'textinput-' + instanceId;
-	errorId = errorId || textInputId;
+	const textInputId = id !== undefined ? id : 'textinput-' + instanceId;
```

Comment:
I thought ESLint would [discourage](https://eslint.org/docs/rules/no-undefined) using `undefined` directly. For some reason that rule is not enabled, but I wonder if we should be extra cautious and convert this check to use `typeof`:

```suggestion
	const textInputId = typeof id !== 'undefined' ? id : 'textinput-' + instanceId;
```

What do you think?